lib/haml_lint/reporter/disabled_config_reporter.rb

```diff
@@ -65,7 +65,7 @@ def finished_file(file, lints)
 
       if lints.any?
         lints.each do |lint|
-          linters_with_lints[lint.linter.name] << lint.filename
+          linters_with_lints[lint.linter.name] |= [lint.filename]
```

Comment:
@trotzig You're thinking of `||=` which is the "memoization method".

This is equivalent to  `linters_with_lints[lint.linter.name] = linters_with_lints[lint.linter.name] | [lint.filename]` which uses [the `#|` method](http://ruby-doc.org/core-2.4.1/Array.html#method-i-7C).

This is very elegant! I actually prefer this over using a Set. I never think to use the `#|` method, so I just reached for the heavy Set.

I can take my tests and drop them on top of this commit. What do you think @trotzig?